test/javascript/explore-data/not-quite-binary-search.spec.js

```diff
@@ -0,0 +1,221 @@
+/**
+ * A building has 100 floors. One of the floors is the highest floor an
+ * egg can be dropped from without breaking.
+ *
+ * If an egg is dropped from above that floor, it will break. If it is
+ * dropped from that floor or below, it will be completely undamaged and
+ * you can drop the egg again.
+ *
+ * Given two eggs, find the highest floor an egg can be dropped from
+ * without breaking, with as few drops as possible.
+ */
+var debug = function() {
+  // console.log(arguments);
+  return;
+};
+
+var Game = function(options) {
+  this.topFloor = (options && typeof options.topFloor === "number") ? options.topFloor : 100;
+  this.eggs = (options && typeof options.eggs === "number") ? options.eggs : 2;
+  this.itterations = 0;
+  this.floorWhereEggBreaks = (options && typeof options.floorWhereEggBreaks === "number") ? options.floorWhereEggBreaks : Math.ceil(Math.random() * this.topFloor);
+  return this;
+};
+
+/**
+ * Proof
+ *
+ * 100 floors:
+ *   1: try each 10 floors
+ *   2: try each floor
+ * = 10, 20, 30, 40, 50, 60, 70, 80, 90, x1, x2, x3, x4, x5, x6, x7, x8, x9
+ * = 17
+ *
+ * 100 floors:
+ *   1: try each 20 floors
+ *   2: try each floor
+ * = 20, 40, 60, 80, x1, x2, x3, x4, x5, x6, x7, x8, x9, x10, x11, x12, x13, x14, x15, x16, x17, x18, x19
+ * = 24
+ *
+ * 100 floors:
+ *   1: try each 5 floors
+ *   2: try each floor
+ * = 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55, 60, 65, 70, 75, 80, 85, 90, 95, x+1, x+2, x+3, x+4
+ * = 24
+ *
+ * 100 floors:
+ *   1: try each 9 floors
+ *   2: try each floor
+ * = 19
+ *
+ *
+ * 48 floors:
+ *  8 x 8
+ *
+ * @param  {[type]} game [description]
+ * @return {[type]}      [description]
+ */
+var findMaxSafeFloor = function(game) {
+  game.multiplier = Math.ceil(Math.sqrt(game.topFloor));
+  var multiplier = game.multiplier * (game.eggs - 1);
+  var currentFloor = 0;
+
+  while (game.eggs && multiplier) {
+    debug("Series: " + multiplier);
+    for (var i = currentFloor + multiplier; i <= game.topFloor; i += multiplier) {
+      debug(" Dropping at ", i);
+      game.itterations += 1;
+
+      if (game.floorWhereEggBreaks <= i) {
+        debug("  Shplat egg " + game.eggs + " broke at " + i + " (" + game.floorWhereEggBreaks + ")");
```

Comment:
for your debug, It might be easier to read with `` instead of +
"  Shplat egg " + game.eggs + " broke at " + i + " (" + game.floorWhereEggBreaks + ")"
to 
` Shplat egg ${game.eggs} broke at ${i} (${game.floorWhereEggBreaks})`

Reply:
Thanks! I agree that would be easier to read, i'll check with the labs to see if they have a modern version of node and then update the project to use es6